test/test_torch.py

```diff
@@ -194,6 +194,24 @@ def test_sinh(self):
     def test_lgamma(self):
         self._testMathByName('lgamma')
 
+    def test_digamma(self):
+        y = torch.Tensor([-10, 0])
+        x = torch.Tensor([-0.1, 3, 999])
```

Comment:
You could also check against hard-coded values generated by `scipy.special.digamma()`, e.g.:
```py
>>> import numpy as np
>>> import scipy.special
>>> x = 10.0 ** np.arange(-5, 5, 0.5)
>>> x
array([  1.00000000e-05,   3.16227766e-05,   1.00000000e-04,
         3.16227766e-04,   1.00000000e-03,   3.16227766e-03,
         1.00000000e-02,   3.16227766e-02,   1.00000000e-01,
         3.16227766e-01,   1.00000000e+00,   3.16227766e+00,
         1.00000000e+01,   3.16227766e+01,   1.00000000e+02,
         3.16227766e+02,   1.00000000e+03,   3.16227766e+03,
         1.00000000e+04,   3.16227766e+04])
>>> scipy.special.digamma(x)
array([ -1.00000577e+05,  -3.16233538e+04,  -1.00005771e+04,
        -3.16285436e+03,  -1.00057557e+03,  -3.16799792e+02,
        -1.00560885e+02,  -3.21491437e+01,  -1.04237549e+01,
        -3.31321826e+00,  -5.77215665e-01,   9.84925052e-01,
         2.25175259e+00,   3.43798293e+00,   4.60016185e+00,
         5.75488076e+00,   6.90725520e+00,   8.05888970e+00,
         9.21029037e+00,   1.03616171e+01])
```